### PR TITLE
[RLlib] Initialize the default view requirements when using the learner group api

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: Avnish <avnishnarayan@gmail.com>

## Why are these changes needed?
This will prevent users from having to implement the policy api except for when they write a trajectory post processor.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
Need to add a test where we construct a policy using a learner group api but without declaring the loss function like we normally have been doing. the check that all the additional keys that we have asked for exist in the sample batch that is returned from the sampler?

File: rllib/policy/eager_tf_policy_v2.py [hunk 1]
```diff
@@ -178,7 +178,19 @@ def loss(
...
+        # Under the new _enable_learner_api the loss function still gets called in order
+        # to initialize the view requirements of the sample batches that are returned by
+        # the sampler. In this case, we don't actually want to compute any loss, however
+        # if we access the keys that are needed for a forward_train pass, then the
+        # sampler will include those keys in the sample batches it returns. This means
+        # that the correct sample batch keys will be available when using the learner
+        # group API.
+        if self.config._enable_learner_api:
+            for k in model.input_specs_train():
+                train_batch[k]
```
Comment (kouroshHakha) [inline rllib/policy/eager_tf_policy_v2.py:190]: what is this doing? `train_batch[k]`?

Comment (avnishn) [reply rllib/policy/eager_tf_policy_v2.py:190]: this accesses the key of the train batch, so that train_batch.accessed keys includes k.

in this case the train_batch that is getting accessed is the dummy batch that is being used to initialize the view requirements.

Comment (avnishn) [reply rllib/policy/eager_tf_policy_v2.py:190]: I'll add an additional comment to make it more clear what is happening here.

Comment (kouroshHakha) [reply rllib/policy/eager_tf_policy_v2.py:190]: so with this change, no-one needs to create any policy instance for any of the new algorithms? This is simplifying the migration guide?

Comment (avnishn) [reply rllib/policy/eager_tf_policy_v2.py:190]: yes exactly

**Review by kouroshHakha — APPROVED** [review]

File: rllib/policy/torch_policy_v2.py [hunk 2]
```diff
@@ -237,7 +237,19 @@ def loss(
...
+        # to initialize the view requirements of the sample batches that are returned by
+        # the sampler. In this case, we don't actually want to compute any loss, however
+        # if we access the keys that are needed for a forward_train pass, then the
+        # sampler will include those keys in the sample batches it returns. This means
+        # that the correct sample batch keys will be available when using the learner
+        # group API.
+        if self.config._enable_learner_api:
+            for k in model.input_specs_train():
+                train_batch[k]
+            return torch.Tensor([0.0])
```
Comment (gjoliver) [inline rllib/policy/torch_policy_v2.py:250]: why return 0.0 tensor? why not return None?

Comment (avnishn) [reply rllib/policy/torch_policy_v2.py:250]: I did some more digging, and we can choose to return none over here since the loss is not used in any way when initializing view requirements. 

I'll go ahead and return instead None type.

Comment (avnishn) [reply rllib/policy/torch_policy_v2.py]: fixed.

**Review by gjoliver — APPROVED** [review]
